packages/sheet-permission-ui/src/command/range-protection.command.ts

```diff
@@ -0,0 +1,374 @@
+/**
+ * Copyright 2023-present DreamNum Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ICommand, Workbook } from '@univerjs/core';
+import { CommandType, ICommandService, IUndoRedoService, IUniverInstanceService, Rectangle, sequenceExecute, UniverInstanceType } from '@univerjs/core';
+import type { ISelectionProtectionRule } from '@univerjs/sheets-selection-protection';
+import { AddSelectionProtection, DeleteSelectionProtection, SelectionProtectionRuleModel, SetSelectionProtection } from '@univerjs/sheets-selection-protection';
+import { AddWorksheetProtectionMutation, DeleteWorksheetProtectionMutation, SelectionManagerService, SetWorksheetProtectionMutation, WorksheetProtectionRuleModel } from '@univerjs/sheets';
+import { UnitObject } from '@univerjs/protocol';
+import { SheetPermissionOpenPanelOperation } from '../operation/sheet-permission-open-panel.operation';
+import { SheetPermissionPanelModel } from '../service/sheet-permission-panel.model';
+import { DeleteWorksheetProtectionCommand } from './worksheet-protection.command';
+import type { IAddRangeProtectionParams, IDeleteRangeProtectionParams, ISetProtectionParams, ISetRangeProtectionParams } from './type';
+
+export const AddRangeProtectionFromToolbarCommand: ICommand = {
```

Comment:
For the name of the command you choosed `range protection`. That's good. Please keep your naming consistent in different modules if there are dealing with the same thing.